From 0008978ea35b6bf97998a71ad6558d2a7b999890 Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Mon, 06 Feb 2023 09:13:59 +0200
Subject: [PATCH] [SW-115382] gaudi2: pll: change sram pll to 350MHz

As part of hl_dev_pll_init, various plls are initialized.
But SRAM PLL is configured to 400MHz instead of 350MHz.
corrected SRAM PLL coefficients to reflect POR frequency table

Currently this code is used only for erom but in upcoming
it will be re-used for pll init at ppboot stage as well

Change-Id: I9327a59583d78ee18d5ab77e0a18401b50576b34
Signed-off-by: Sailaja <sbandarupalli@habana.ai>
---

diff --git a/arch/gaudi2/common/asic.c b/arch/gaudi2/common/asic.c
index 666a064..aadedd7 100644
--- a/arch/gaudi2/common/asic.c
+++ b/arch/gaudi2/common/asic.c
@@ -170,7 +170,7 @@
 	}, {
 		.name = "DCORE2_SRAM",
 		.base = mmDCORE2_XBAR_BANK_PLL_CTRL_BASE,
-		.nr = 1, .nf = 64, .od = 4, .nb = 2,	/* 400 */
+		.nr = 1, .nf = 63, .od = 3, .nb = 3,	/* 350 */
 		.div_factor = { 0, -1, -1, -1 },
 		.div_sel = { HL_PLL_PLL_CLK,
 			     HL_PLL_REF_CLK,
@@ -180,7 +180,7 @@
 	}, {
 		.name = "DCORE3_SRAM",
 		.base = mmDCORE3_XBAR_BANK_PLL_CTRL_BASE,
-		.nr = 1, .nf = 64, .od = 4, .nb = 2,	/* 400 */
+		.nr = 1, .nf = 63, .od = 3, .nb = 3,	/* 350 */
 		.div_factor = { 0, -1, -1, -1 },
 		.div_sel = { HL_PLL_PLL_CLK,
 			     HL_PLL_REF_CLK,
